src/Database/Database.php

```
@@ -8518,4 +8978,183 @@ protected function encodeSpatialData(mixed $value, string $type): string
+                    try {
+                        $rollbackOperation();
+                    } catch (\Throwable $e) {
+                        // Silent rollback - errors are swallowed
+                    }
+                } else {
+                    // Regular mode: rollback throws on failure
+                    try {
+                        $rollbackOperation();
+                    } catch (\Throwable $e) {
+                        throw new DatabaseException(
+                            "Failed to persist metadata after retries and cleanup failed for {$operationDescription}: " . $e->getMessage() . ' | Cleanup error: ' . $rollbackException->getMessage(),
```

PHPMD (2.15.0)

Avoid unused local variables such as '$rollbackException'. (undefined)

(UnusedLocalVariable)